src/scheduler/check.cpp

```diff
@@ -1143,128 +1263,89 @@ check_avail_resources(schd_resource *reslist, resource_req *reqlist,
-					continue;
-
-				res->name = resreq->name;
-				res->def = resreq->def;
-			}
+			num_chunk = do_compare(res, resreq, flags, fail_code, err);
 
-			if (res->indirect_res != NULL) {
-				res = res->indirect_res;
-			}
-
-			if (res->type.is_non_consumable && !(flags & ONLY_COMP_CONS)) {
-				if (!compare_non_consumable(res, resreq)) {
-					fail = 1;
+			if (num_chunk == 0) {
+				any_fail = 1;
+				if (flags & RETURN_ALL_ERR) {
 					if (err != NULL) {
-						const char *requested;
-						set_schd_error_codes(err, NOT_RUN, fail_code);
-						err->rdef = res->def;
-						requested = res_to_str_r(resreq, RF_REQUEST, resbuf1, sizeof(resbuf1));
-						snprintf(buf, sizeof(buf), "(%s != %s)",
-							requested,
-							res_to_str_r(res, RF_AVAIL, resbuf2, sizeof(resbuf2)));
-						set_schd_error_arg(err, ARG1, buf);
-						/* Set arg2 for vnode/host resource. In case of preemption, arg2 is used to cull
-						 * the list of running jobs
-						 */
-						if ((res->def == getallres(RES_HOST)) || (res->def == getallres(RES_VNODE)))
-							set_schd_error_arg(err, ARG2, requested);
+						err->next = new_schd_error();
+						if (err->next == NULL)
+							return 0;
+						prev_err = err;
+						err = err->next;
 					}
-				}
+				} else
+					break;
 			}
-			else if (res->type.is_consumable && !(flags & ONLY_COMP_NONCONS)) {
-				if (flags & COMPARE_TOTAL)
-					avail = res->avail;
-				else
-					avail = dynamic_avail(res);
+		}
+	}
 
-				if (avail == SCHD_INFINITY_RES && (flags & UNSET_RES_ZERO))
-					avail = 0;
+	if (any_fail)
+		num_chunk = 0;
 
-				/*
-				 * if there is an infinite amount available or we are requesting
-				 * 0 amount of the resource, we do not need to check if any is
-				 * available
-				 */
-				if (avail != SCHD_INFINITY_RES && resreq->amount != 0) {
-					if (avail < resreq->amount) {
-						fail = 1;
-						if (err != NULL) {
-							set_schd_error_codes(err, NOT_RUN, fail_code);
-							err->rdef = res->def;
-
-							res_to_str_r(resreq, RF_REQUEST, resbuf1, sizeof(resbuf1));
-							res_to_str_c(avail, res->def, RF_AVAIL, resbuf2, sizeof(resbuf2));
-							if ((flags & UNSET_RES_ZERO) && res->avail == SCHD_INFINITY_RES)
-								res_to_str_c(0, res->def, RF_AVAIL, resbuf3, sizeof(resbuf3));
-							else
-								res_to_str_r(res, RF_AVAIL, resbuf3, sizeof(resbuf3));
-							snprintf(buf, sizeof(buf), "(R: %s A: %s T: %s)", resbuf1, resbuf2, resbuf3);
-							set_schd_error_arg(err, ARG1, buf);
+	if (prev_err != NULL && (flags & RETURN_ALL_ERR)) {
+		if (prev_err != NULL) {
+			free_schd_error(err);
+			prev_err->next = NULL;
+		}
+	}
 
-						}
-					}
-					else {
-						cur_chunk = avail / resreq->amount;
-						if (cur_chunk < num_chunk || num_chunk == SCHD_INFINITY)
-							num_chunk = cur_chunk;
-					}
-				}
-			}
-			if(fail && (flags & RETURN_ALL_ERR)) {
-				fail = 0;
-				any_fail = 1;
+	return num_chunk;
+}
+
+// overloaded check_avail_resources() that checks all resources
```

Comment:
The arguments to both functions are the same (except the missing one).  I figured it would be redundant.  I can write one if you want.

Reply:
I think it would be helpful, otherwise the doxygen documentation will not report overloaded functions

Reply:
Really? I thought doxygen will report the function but it wouldn't display anything else because there is no header.

Reply:
hmm, ya you are probably right, so it will report the function name, but not describe the arguments .. anyways, I'm not particular about this, I'll leave it to you to decide.

Reply:
I'll add a @brief to the overload and point it to look at the main function for the arguments.